lib/internal/abort_controller.js

```diff
@@ -197,7 +197,10 @@ class AbortSignal extends EventTarget {
   static any(signals) {
     validateAbortSignalArray(signals, 'signals');
     const resultSignal = createAbortSignal({ composite: true });
-    const resultSignalWeakRef = new WeakRef(resultSignal);
+    let resultSignalWeakRef;
+    if (signals.length) {
+      resultSignalWeakRef ??= new WeakRef(resultSignal);
+    }
```

Comment:
I think using `&&` would be fine here.
```suggestion
    const resultSignalWeakRef = signals.length && new WeakRef(resultSignal);
```
If you disagree, we can drop the nullish assignement
```suggestion
    let resultSignalWeakRef;
    if (signals.length) {
      resultSignalWeakRef = new WeakRef(resultSignal);
    }
```

Reply:
Changed to an early-exit based on @Linkgoron comment below, so this is now a const as was before, resolving 🙂